drivers: sensor: lm73c: sample: Added TI LM73C sensor driver

riedlse opened the pull request:
New pull request replaces #31064 (mistake on my part created a new branch)

Driver for TI LM73C chip -40 to 150 C I2C device. Fixed all issues in #31064 and a few new ones I learned from the
other drivers I am working on.

Used separate commits for driver and sample code

Signed-off-by: Steven Riedl steve@serconsultingllc.com

dcpleung committed:
x86: define dynamic thread object alignment

x86 and x86_64 require certain alignment in the k_thread struct
since the buffer to save/restore FPU/SSE registers requires
strict alignment.

Fixes #29589
Fixes #29629

Signed-off-by: Daniel Leung <daniel.leung@intel.com>

mbolivar-nordic commented:
> New pull request replaces #31064 (mistake on my part created a new branch)

Could you please just force-push the new commits into the old PR instead?

riedlse committed:
drivers: sensor: lm73c: sample: Added TI LM73C sensor driver

No changes, just trying to get commit messages correct

Signed-off-by: Steven Riedl <steve@serconsultingllc.com>

riedlse commented:
> > New pull request replaces #31064 (mistake on my part created a new branch)
> 
> Could you please just force-push the new commits into the old PR instead?

I removed the old branch, still learning a bit on adding to a large project.

riedlse committed:
drivers: sensor: lm73c: sample: Added TI LM73C sensor driver

Removed unused config from Kconfig, needed to switch to
config from menuconfig

Signed-off-by: Steven Riedl <steve@serconsultingllc.com>

riedlse commented:
Don't have time to keep going through this now, may resubmit at a later date.
